### PR TITLE
Handle API errors correctly

### PR DESCRIPTION
**Pull request by nwalters512**
This wraps all route handlers with a function that will catch any errors and send them through `express`.

### CODE REVIEW

File: api/queues.js [hunk 1]
```diff
@@ -49,7 +53,7 @@ router.post(
     ]),
     failIfErrors,
   ],
-  (req, res, _next) => {
+  (req, res, next) => {
```
Comment (genevievehelsel) [inline api/queues.js]: is there a reason this was changed from _next to next?

File: api/queues.js [hunk 2]
```diff
@@ -61,15 +65,18 @@ router.post(
       createdByUserId: res.locals.userAuthn.id,
     })
 
-    queue.save().then(newQueue => res.status(201).json(newQueue))
+    queue
+      .save()
+      .then(newQueue => res.status(201).json(newQueue))
+      .catch(next)
```
Comment (genevievehelsel) [inline api/queues.js]: why is this post catching instead of using the wrapper?

File: api/queues.js (hunk 1 above)
Comment (nwalters512) [reply api/queues.js]: The underscore was used to indicate an unused param, but it's actually used now.

File: api/queues.js (hunk 2 above)
Comment (nwalters512) [reply api/queues.js]: The wrapper is really only for `async` stuff, this is using promises directly so it's better to use `catch` directly.

**Pull request merged**